File: c/day1_part2_2015.c
#include <stdio.h>

int main() {
    FILE *fp;
    char ch;
    int floor = 0;
    int position = 0;
    int basement_position = 0;

    fp = fopen("input.txt", "r");

    if (fp == NULL) {
        printf("Error opening file.\n");
        return 1;
    }

    while ((ch = fgetc(fp)) != EOF) {
        position++;
        if (ch == '(') {
            floor++;
        } else if (ch == ')') {
            floor--;
            if (floor == -1 && basement_position == 0) {
                basement_position = position;
            }
        }
    }

    fclose(fp);

    printf("%d\n", floor);
    printf("%d\n", basement_position);

    return 0;
}

File: swift/day1_part2_2015.swift
import Foundation

let input = try String(contentsOfFile: "input.txt")
var floor = 0
var position = 0

for char in input {
    position += 1
    if char == "(" {
        floor += 1
    } else if char == ")" {
        floor -= 1
    }
    
    if floor == -1 {
        break
    }
}

print(position)